packages/tracer/tests/unit/ProviderService.test.ts

```diff
@@ -1,102 +1,94 @@
-/**
- * Test ProviderService class
- *
- * @group unit/tracer/providerservice
- */
 import { channel } from 'node:diagnostics_channel';
 import http from 'node:http';
 import https from 'node:https';
 import { URL } from 'node:url';
 import { addUserAgentMiddleware } from '@aws-lambda-powertools/commons';
 import { DynamoDBClient } from '@aws-sdk/client-dynamodb';
-import {
-  Segment,
-  Subsegment,
-  captureAWS,
-  captureAWSClient,
-  captureAWSv3Client,
-  captureAsyncFunc,
-  captureFunc,
-  captureHTTPsGlobal,
-  getNamespace,
-  getSegment,
-  setContextMissingStrategy,
-  setDaemonAddress,
-  setLogger,
-  setSegment,
-} from 'aws-xray-sdk-core';
+import xraySDK from 'aws-xray-sdk-core';
+import { beforeEach, describe, expect, it, vi } from 'vitest';
 import { ProviderService } from '../../src/provider/ProviderService.js';
 import type { HttpSubsegment } from '../../src/types/ProviderService.js';
 import { mockFetch } from '../helpers/mockRequests.js';
 
-jest.mock('aws-xray-sdk-core', () => ({
-  ...jest.requireActual('aws-xray-sdk-core'),
-  captureAWS: jest.fn(),
-  captureAWSClient: jest.fn(),
-  captureAWSv3Client: jest.fn(),
-  captureAsyncFunc: jest.fn(),
-  captureHTTPsGlobal: jest.fn(),
-  captureFunc: jest.fn(),
-  getNamespace: jest.fn(),
-  getSegment: jest.fn(),
-  setContextMissingStrategy: jest.fn(),
-  setDaemonAddress: jest.fn(),
-  setLogger: jest.fn(),
-  setSegment: jest.fn(),
+const { Segment, Subsegment } = xraySDK;
+const mocks = vi.hoisted(() => ({
+  captureAWS: vi.fn(),
+  captureAWSClient: vi.fn(),
+  captureAWSv3Client: vi.fn(),
+  captureAsyncFunc: vi.fn(),
+  captureHTTPsGlobal: vi.fn(),
+  captureFunc: vi.fn(),
+  getNamespace: vi.fn(),
+  getSegment: vi.fn(),
+  setContextMissingStrategy: vi.fn(),
+  setDaemonAddress: vi.fn(),
+  setLogger: vi.fn(),
+  setSegment: vi.fn(),
 }));
 
-jest.mock('@aws-lambda-powertools/commons', () => ({
-  ...jest.requireActual('@aws-lambda-powertools/commons'),
-  addUserAgentMiddleware: jest.fn(),
+vi.mock('aws-xray-sdk-core', async (importOriginal) => ({
+  default: {
+    ...(
+      await importOriginal<{
+        default: typeof import('aws-xray-sdk-core');
+      }>()
+    ).default,
+    ...mocks,
+  },
+}));
```

Comment:
wow, this is an interesting level of sorcery

Reply:
Yes, it's not very intuitive.

Here are [the docs for the Vitest mocking feature](https://vitest.dev/api/vi.html#mock-modules). Basically we are using Vitest to hijack the module resolution for a specific module and returning our own object.

In doing so however, we are not replacing the entire module but only some of it. To do so, we are importing the original module and destructuring its content in the object we return. Next we overwrite some of the properties that we want to provide a mock for (see `mocks` object).

This specific module has also an added complexity which is the fact that it only provides a `default` export, which is why we need to nest the mock object within a `default` property.